### DISCUSSION TITLE
Handling intermittent replication failures

### DISCUSSION THREAD

**Discussion by BrandonWeng** [goharbor/harbor#15380] [Q&A]
We observed a transient failure with our Redis instance that caused some replication jobs to not get queued for an event based replication rule. The straight forward approach seems to be running the replication on everything, but it seems like the job is FIFO so that means a full replication may cause the replication to be significantly behind. 

1. Is there a way to determine the delta between two repositories (i.e which image is missing exactly)?
2. Is there an alternative to running the replicate rule on the entire project if we can't determine which images are missing exactly? 

Thanks in advance!

**Comment by wy65701436** [comment]
cc @steven-zou we have fixed some issues related on jobservice, what's the Harbor version? 

Is there a way to determine the delta between two repositories (i.e which image is missing exactly)?
Currently, you have to use the API to list all the repository & artifact from both source & target, then do comparation.

Is there an alternative to running the replicate rule on the entire project if we can't determine which images are missing exactly?
Just specify ** in the resource filter to match all, it will replicate the entire project.

**Comment by BrandonWeng** [comment]
@wy65701436 sorry for the late response, thanks for the useful comments! We'll have to try to create a tool that compares the two harbor instances. 

One of the issues is that our Harbor registry is growing quite a bit and a full replication took several hours to complete, even though only a handful of images are missing in the follower instance. 

We're running `Version v2.2.2-56d7937f`
